### PR TITLE
azblob: Added support for getSASURL from specialized clients

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

This PR contains the following fixes/features - 
1. Expose GetSASURL on specialized clients - This will let customers call getSASUrl without creating outer BlobClient
2. client.BlobClient() should pass the credential as well.

### CODE REVIEW

File: sdk/storage/azblob/blob/client_test.go [hunk 1]
```diff
@@ -3655,39 +3655,40 @@ func (s *BlobUnrecordedTestsSuite) TestSASURLBlobClient() {
...
-	opts := blob.GetSASURLOptions{StartTime: &start}
-
-	// GetSASURL fails (with MissingSharedKeyCredential) because blob client is created without credentials
-	_, err = bbClient.BlobClient().GetSASURL(permissions, expiry, &opts)
-	_require.Equal(err, bloberror.MissingSharedKeyCredential)
-}
+//func (s *BlobUnrecordedTestsSuite) TestNoSharedKeyCredError() {
+//	_require := require.New(s.T())
+//	testName := s.T().Name()
+//
```
Comment (vibhansa-msft) [inline sdk/storage/azblob/blob/client_test.go:3661]: If this is dead code then lets remove this.

File: sdk/storage/azblob/appendblob/client.go [hunk 2]
```diff
@@ -338,6 +339,12 @@ func (ab *Client) CopyFromURL(ctx context.Context, copySource string, o *blob.Co
 	return blob.CopyFromURLResponse{}, errors.New("operation will not work on this blob type. CopyFromURL works only with block blob")
 }
 
+// GetSASURL is a convenience method for generating a SAS token for the currently pointed at append blob.
+// It can only be used if the credential supplied during creation was a SharedKeyCredential.
+func (ab *Client) GetSASURL(permissions sas.BlobPermissions, expiry time.Time, o *blob.GetSASURLOptions) (string, error) {
```
Comment (vibhansa-msft) [inline sdk/storage/azblob/appendblob/client.go:344]: This has been added to blockblob and pageblob. Do we need the same for AppendBlob as well?

Comment (tanyasethi-msft) [reply sdk/storage/azblob/appendblob/client.go:344]: This has been added to all specialized clients - including Append Blob.
See AppendBlob implementation of GetSASUrl here - https://github.com/Azure/azure-sdk-for-go/blob/b268345f3c913cfcd8d6138b15e3e1445ec9dabe/sdk/storage/azblob/appendblob/client.go#L344

File: sdk/storage/azblob/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@
 ### Bugs Fixed
 
 * Fixed case in Blob Batch API when blob path has / in it. Fixes [#21649](https://github.com/Azure/azure-sdk-for-go/issues/21649).
+* Exposed GetSASURL from specialized clients
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/CHANGELOG.md:12]: This should be added in the `Features Added` section IMO.

File: sdk/storage/azblob/CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,8 @@
...
+* Fixed SharedKeyMissingError when using client.BlobClient() method
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/CHANGELOG.md:13]:
```suggestion
* Fixed SharedKeyMissingError when using client.BlobClient().GetSASURL() method
```

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**